src/cpp/lib/core/Events/eventwidget.cpp

```diff
@@ -41,11 +45,25 @@ void EventWidget::event(QEvent* event) {
       std::string eventTypeString = subscribedEvents.at(evtType);
       Napi::Env env = this->emitOnNode.Env();
       Napi::HandleScope scope(env);
+      std::vector<napi_value> args;
+
+      switch (evtType) {
```

Comment:
I dont think we should have a switch case here. 
The original idea is to do something like this:
```
win.addEventListener(BaseWidgetEvents.KeyRelease, nativeEvent => {
  const keyEvt = new QKeyEvent(nativeEvent);
  const text = keyEvt.text();
});```

This allows us to implement event wrappers independently. 
We could even implement event wrappers as a plugin.

So I would recommend removing this switch case. But rest all looks good 👍

Reply:
So that's how it works! The casting of native events on JS side works great, thank you. I reverted eventwidget.cpp back to original state and now all that remains is the small extensions to QKeyEvent.